WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt

```diff
@@ -0,0 +1,145 @@
+    isNameInputHidden: Boolean,
+    emailAndNameSelected: (String, String) -> Unit
+) {
+    val (layout, emailEditText, nameEditText) =
+            supportIdentityInputDialogLayout(context, isNameInputHidden, email, name)
+
+    val dialog = AlertDialog.Builder(context, R.style.Calypso_Dialog)
+            .setView(layout)
+            .setPositiveButton(android.R.string.ok, null)
+            .setNegativeButton(android.R.string.cancel, null)
+            .create()
+    dialog.setOnShowListener {
+        val button = (dialog as AlertDialog).getButton(AlertDialog.BUTTON_POSITIVE)
+        button.setOnClickListener { _ ->
+            val newEmail = emailEditText.text.toString()
+            val newName = nameEditText.text.toString()
+            if (validateEmail(newEmail)) {
+                emailAndNameSelected(newEmail, newName)
+                dialog.dismiss()
+            } else {
+                emailEditText.error = context.getString(R.string.invalid_email_message)
+            }
+        }
+    }
+    dialog.show()
+}
+
+/**
+ * This is a helper function that inflates the support identity dialog layout.
+ *
+ * @param context Context to use to inflate the layout
+ * @param isNameInputHidden Whether the name EditText should be visible or not
+ * @param emailSuggestion Initial value for the email EditText
+ * @param nameSuggestion Initial value for the name EditText
+ *
+ * @return a Triple with layout View, email EditText and name EditText
+ */
+private fun supportIdentityInputDialogLayout(
+    context: Context,
+    isNameInputHidden: Boolean,
+    emailSuggestion: String?,
+    nameSuggestion: String?
+): Triple<View, EditText, EditText> {
+    val layout = LayoutInflater.from(context).inflate(R.layout.support_email_and_name_dialog, null)
+
+    val messageText = layout.findViewById<TextView>(R.id.support_identity_input_dialog_message)
+    val message = if (isNameInputHidden) {
+        R.string.support_identity_input_dialog_enter_email
+    } else {
+        R.string.support_identity_input_dialog_enter_email_and_name
+    }
+    messageText.setText(message)
+
+    val emailEditText = layout.findViewById<EditText>(R.id.support_identity_input_dialog_email_edit_text)
+    emailEditText.setText(emailSuggestion)
+    emailEditText.setSelection(0, emailSuggestion?.length ?: 0)
+
+    val nameEditText = layout.findViewById<EditText>(R.id.support_identity_input_dialog_name_edit_text)
+    nameEditText.setText(nameSuggestion)
+    nameEditText.visibility = if (isNameInputHidden) GONE else View.VISIBLE
+
+    return Triple(layout, emailEditText, nameEditText)
+}
+
+/**
+ * This is a helper function to returns suggested email and name values to be used in the support identity dialog.
+ *
+ * @param account WordPress.com account
+ * @param selectedSite Selected site of the user which will be used if the [account] is null
+ *
+ * @return a Pair with email and name suggestion
+ */
+private fun supportEmailAndNameSuggestion(
```

Comment:
Sorry to be nitpicky about the naming again but it's not clear what this method does (sets vs. gets).

Reply:
Changed to `getSupportEmailAndNameSuggestion` in da3a8a90aca14e00c356d258387ee7663f0734e5.